pkg/virt-handler/vm.go

```diff
@@ -2876,3 +2898,18 @@ func (d *VirtualMachineController) reportTargetTopologyForMigratingVMI(vmi *v1.V
 	vmi.Status.MigrationState.TargetNodeTopology = string(topology)
 	return nil
 }
+
+func (d *VirtualMachineController) handleMigrationAbort(vmi *v1.VirtualMachineInstance, client cmdclient.LauncherClient) error {
+	if vmi.Status.MigrationState.AbortStatus == v1.MigrationAbortInProgress {
+		return nil
+	}
+
+	err := client.CancelVirtualMachineMigration(vmi)
+	if err != nil && err.Error() != migrations.CancelMigrationFailedVmiNotMigratingErr {
```

Comment:
won't we hit this error 100% of the time? 
What is the scenario that you are thinking about in this case? In general, how could the virt-launcher start an actual migration without having a destination pod? I mean, `isPreMigrationSource` not checking for a `TargetNodeAddress` for a reason, but it's a must for creating a migration proxy.

Reply:
I'm not entirely sure I understand your concern here.

> won't we hit this error 100% of the time?

I don't see why. This error will appear only if the source virt-launcher does not know of any migration going on, as can be seen here:

https://github.com/kubevirt/kubevirt/blob/c6235d06881db870893e5121803731c5c76aec29/pkg/virt-launcher/virtwrap/live-migration-source.go#L484-L488

So we would hit this error only if we have an error that is _different than_ `CancelMigrationFailedVmiNotMigratingErr`.

> What is the scenario that you are thinking about in this case?

The scenario is that we start a migration then cancel it right away. This is actually exactly the scenario that occur in the functional test that's added to this PR. The functional test failed without this condition, as what happened was the following:

1. VMIM object is posted, migration "starts"
2. VMIM is being deleted right away (before any of our components have actually started the migration process)
3. `handleMigrationAbort()` is called, which in turn calls `CancelVirtualMachineMigration()`
4. `CancelVirtualMachineMigration()` returns an error from virt-launcher that says "I'm not aware of any migration going on"
5. Test fails, or in reality, controller keeps trying to perform the same task of cancelling the migration that failed with an error. Once the migration actually starts by virt-launcher, `CancelVirtualMachineMigration()` will succeed and the migration would be cancelled.

In other words - this condition is to ensure we're able to cancel a migration before it actually starts and recognized by virt-launcher. If the launcher says that it's not aware of any migration occurring, we can simply remove the VMIM's finalizer and delete it.

Reply:
I feel like I don't understand why you need to eat the error. We should be eventually consistent here.

Reply:
I would at least log that we are attempting cancelation if the above handling stays.

Reply:
yup, I agree. we should log that there was an attempt to cancel the migration. It will help us understand the context better.

Reply:
Added a log entry :ok_hand: